File: Data Structures & Algorithms/permutation-string/submission-10.cpp
class Solution {
public:
    bool checkInclusion(std::string s1, std::string s2) {
        sort(s1.begin(), s1.end());

        for (int i = 0; i <= s2.size() - s1.size(); i++) {
            string subStr = s2.substr(i, s1.size());
            sort(subStr.begin(), subStr.end());
            if (subStr == s1) {
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-10.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        List<int[]> A = new List<int[]>();
        for (int i = 0; i < nums.Length; i++) {
            A.Add(new int[]{nums[i], i});
        }

        A.Sort((a, b) => a[0].CompareTo(b[0]));

        int i = 0, j = nums.Length - 1;
        while (i < j) {
            int cur = A[i][0] + A[j][0];
            if (cur == target) {
                return new int[]{
                    Math.Min(A[i][1], A[j][1]), 
                    Math.Max(A[i][1], A[j][1])
                };
            } else if (cur < target) {
                i++;
            } else {
                j--;
            }
        }
        return new int[0];
    }
}